packages/compiler-sfc/src/parse.ts

```diff
@@ -95,11 +96,18 @@ export function parse(
     sourceRoot = '',
     pad = false,
     ignoreEmpty = true,
-    compiler = CompilerDOM
+    compiler = CompilerDOM,
+    defaultScriptLang
   }: SFCParseOptions = {}
 ): SFCParseResult {
   const sourceKey =
-    source + sourceMap + filename + sourceRoot + pad + compiler.parse
+    source +
+    sourceMap +
+    filename +
+    sourceRoot +
+    pad +
+    compiler.parse +
+    (defaultScriptLang ?? '')
```

Comment:
IMO nullish coalescing operator (`??`) doesn't match the codebase style. Its better to use `||` here.

Reply:
thanks